### PR TITLE
Improve the performance of prefix list after fix #4668

### LINKED ISSUE #4668: New version of seaweedfs can't list directory using s3 api
Sponsors SeaweedFS via Patreon https://www.patreon.com/seaweedfs
Report issues here. Ask questions here https://stackoverflow.com/questions/tagged/seaweedfs
Please ask questions in https://github.com/seaweedfs/seaweedfs/discussions

**Describe the bug**
Let me show you the behaviour, the client I used was minio client. with seaweed version 3.53 and 3.43
```shell
# using weed version 3.53
$ mc ls weed-admin/images/reverse_bag_images/
# NO OUTPUT FOR THE ABOVE COMMAND
$ mc ls weed-admin/images/reverse_bag_images
[2023-07-12 12:06:09 +07]     0B reverse_bag_images/

# revert update to version 3.43
$ mc ls weed-admin/images/reverse_bag_images
[2023-07-12 12:13:17 +07]     0B 2023-07-03/
[2023-07-12 12:13:17 +07]     0B 2023-07-04/
$ mc ls weed-admin/images/reverse_bag_images/
[2023-07-12 12:13:19 +07]     0B 2023-07-03/
[2023-07-12 12:13:19 +07]     0B 2023-07-04/
```
As you can see from the above output, s3 file list was only working when I uses the older version of seaweedfs.

**My story**

I was happily using seaweedfs for a long time without any problem, until yesterday that I try update my servers to newest version, that I encounter this problem. After browsing through the releases changes log, I suspect [this release](https://github.com/seaweedfs/seaweedfs/releases/tag/3.46). My current solution is to revert the update, and continue to use 3.43 for now.

**System Setup**
```shell
$ uname -a
Linux example-storage-2 5.15.0-76-generic #83-Ubuntu SMP Thu Jun 15 19:16:32 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

$ weed version
version 30GB 3.43 3227e4175e2bf8df2ac8aeeff8cf73a819abc5a7 linux amd64

$ /usr/bin/weed master -mdir=/etc/seaweedfs/weed-meta -ip=storage2.example.com -ip.bind=0.0.0.0 -port=9333 -defaultReplication=010 -peers=storage3.example.com:9333,storage4.example.com:9333
$ /usr/bin/weed volume -mserver=storage2.example.com:9333,storage3.example.com:9333,storage4.example.com:9333 -max=0 -ip.bind=0.0.0.0 -port=8081 -dataCenter=main-site -rack=example-storage-2 -dir=/mnt/nvme1/seaweedfs,/mnt/nvme2/seaweedfs,/mnt/nvme3/seaweedfs,/mnt/nvme4/seaweedfs
$ /usr/bin/weed filer -master=storage2.example.com:9333,storage3.example.com:9333,storage4.example.com:9333 -dataCenter=main-site -ip=storage2.example.com -ip.bind=0.0.0.0 -s3 -s3.dataCenter=main-site -s3.config=/etc/seaweedfs/s3-config.json -s3.cert.file=/data/letsencrypt/certificates/example-storage-2.example.com.crt -s3.key.file=/data/letsencrypt/certificates/example-storage-2.example.com.key
```

```shell
$ cat /etc/seaweedfs/filer.toml
```
```toml
# A TOML config file for SeaweedFS filer store

####################################################
# Customizable filer server options
####################################################
[filer.options]
# with http DELETE, by default the filer would check whether a folder is empty.
# recursive_delete will delete all sub folders and files, similar to "rm -Rf"
recursive_delete = false

####################################################
# The following are filer store options
####################################################
# We're forced to use redis sentinel since redis cluster require at least 6 nodes (we have only 4)
[redis3_sentinel]
enabled = true
addresses = ["redis1.storage.example:26379","redis2.storage.example:26379","redis3.storage.example:26379","redis4.storage.example:26379"]
masterName = "mymaster"
username = "default"
password = "RETRACTED"
database = 0
```

**Expected behavior**
I expect that I can use s3 api to list files

### PR DESCRIPTION
**Pull request by lakeland1990**
# What problem are we solving?

after remove compare in pr #4924 , the loop will fetch some objects based on the limit parameter, when limit parameter 
When the limit parameter is set too low and the number of objects is too large, it can cause performance issues.

# How are we solving the problem?

The limit parameter only focuses on the number of results and should not affect the number of entries obtained each time. Modify the List operation of prefixFilterEntries to get a fixed number (500) of entries each time to improve performance. 

# How is the PR tested?

all test passed

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

File: weed/filer/filerstore_wrapper.go [hunk 1]
```diff
@@ -282,7 +282,8 @@ func (fsw *FilerStoreWrapper) prefixFilterEntries(ctx context.Context, dirPath u
 	}
 
 	var notPrefixed []*Entry
-	lastFileName, err = actualStore.ListDirectoryEntries(ctx, dirPath, startFileName, includeStartFile, limit, func(entry *Entry) bool {
+	fetchLimit := int64(500)
```
Comment (kmlebedev) [inline weed/filer/filerstore_wrapper.go]: may be make constant ?

Comment (lakeland1990) [reply weed/filer/filerstore_wrapper.go]: OK，i will set a const

Comment (chrislusf) [reply weed/filer/filerstore_wrapper.go]: how about only set to `FetchLimit` if `limit < FetchLimit`?

Comment (lakeland1990) [reply weed/filer/filerstore_wrapper.go]: > how about only set to `FetchLimit` if `limit < FetchLimit`?

I suppose the best number can only found by benchmark , but set by parameter limit here is not a good idea ,as i mentioned : The limit parameter supposed to limit the number of results and should not affect the number of entries obtained each time.

**Pull request closed without merging**